daemon/cluster/executor/container/controller.go

```diff
@@ -142,7 +142,68 @@ func (r *controller) Start(ctx context.Context) error {
 		return errors.Wrap(err, "starting container failed")
 	}
 
-	return nil
+	// no health check
+	if ctnr.Config == nil || ctnr.Config.Healthcheck == nil {
+		return nil
+	}
+
+	healthCmd := ctnr.Config.Healthcheck.Test
+
+	if len(healthCmd) == 0 {
+		// this field should be filled, even if inherited from image
+		// if it's empty, health check will always be at starting status
+		// so treat it as no health check, and return directly
+		return nil
+	}
+
+	// health check is disabled
+	if healthCmd[0] == "NONE" {
+		return nil
+	}
+
+	// wait for container to be healthy
+	eventq := r.adapter.events(ctx)
+
+	var healthErr error
+	for {
+		select {
+		case event := <-eventq:
```

Comment:
@aaronlehmann @stevvooe could you please help review this event workflow? I think it's worth a careful review to avoid race and other anomalies.